tinygrad/runtime/ops_gpu.py

```diff
@@ -103,4 +103,4 @@ def __call__(self, global_size, local_size, *bufs, wait=False) -> Optional[float
         return None
     return None
 
-GPUBuffer = Compiled(CLBuffer, LinearizerOptions(), OpenCLRenderer, CLProgram, CL.synchronize)
+GPUBuffer = Compiled(CLBuffer, LinearizerOptions(device="GPU", supports_float16=False), OpenCLRenderer, CLProgram, CL.synchronize)
```

Comment:
github's intel GPUs don't implement `half` in their drivers. The extension is automatically disabled.

https://github.com/tinygrad/tinygrad/actions/runs/6508596058/job/17678264259?pr=2046#step:13:117

right now this is a static flag. to support both kinds of GPUs _(even though no fp16 ones will kind of be broken for some functionality)_ this needs to be logic in the code. Either something we pre-compute on device initialization and assign to `supports_float16` in the tinygrad runtime or branching in the kernel itself

Reply:
https://github.com/fangq/mcxcl/issues/24#issuecomment-511815534